src/hooks/usePositions.tsx

```diff
@@ -58,9 +75,73 @@ export const usePositions = (options: OptionsToSearch) => {
 
   React.useEffect(() => {
     if (positionsData) {
-      setData(marshalPositionListData(positionsData.positions, userData?.user))
+      const positionListData = marshalPositionListData(positionsData.positions, userData?.user)
+
+      const fetchUserBalanceWithDecimals = async () => {
+        const uniqueCollateralTokens = uniqBy(positionListData, 'collateralToken')
+
+        const collateralTokensPromises = uniqueCollateralTokens.map(async (position: Position) => {
+          const { collateralToken } = position
+
+          try {
+            const { decimals } = networkConfig.getTokenFromAddress(collateralToken)
+            return {
+              collateralToken,
+              decimals,
+            }
+          } catch (err) {
+            logger.error(err)
+          }
+
+          try {
+            const erc20Service = new ERC20Service(provider, collateralToken)
+            const { decimals } = await erc20Service.getProfileSummary()
+
+            return {
+              collateralToken,
+              decimals,
+            }
+          } catch (err) {
+            logger.error(err)
+          }
+
+          return {
+            collateralToken,
+            decimals: null,
+          }
+        })
+        const collateralTokensResolved = await Promise.all(collateralTokensPromises)
+
+        const positionListDataPromises = positionListData.map(async (position: Position) => {
+          const { collateralToken, userBalance } = position
+
+          const collateralTokenFound = collateralTokensResolved.filter(
+            (collateralTokenInformation) => {
+              return (
+                collateralTokenInformation.collateralToken.toLowerCase() ===
+                collateralToken.toLowerCase()
+              )
+            }
+          )
+
+          const userBalanceWithDecimals =
+            collateralTokenFound.length && collateralTokenFound[0].decimals
+              ? formatBigNumber(userBalance, collateralTokenFound[0].decimals)
+              : userBalance.toString()
+
+          return {
+            ...position,
+            userBalanceWithDecimals,
+          } as PositionWithUserBalanceWithDecimals
+        })
+
+        const positionListDataResolved = await Promise.all(positionListDataPromises)
```

Comment:
Removing `async` from ` positionListData.map(async (position: Position) => {` should lead to removing the `Promise.all`

Reply:
Ohh I forgot to remove it, good catch, thanks